tools/harvester-forwarding/src/functions/compareDefinitions.js

```diff
@@ -0,0 +1,250 @@
+const { app } = require("@azure/functions");
+
+app.http("compareDefinitions", {
+  methods: ["POST"],
+  authLevel: "anonymous",
+  handler: async (request, context) => {
+    try {
+      const {
+        productionDoc,
+        stagingDoc,
+        ignoredKeys = [],
+      } = await request.json();
+
+      if (!productionDoc || !stagingDoc) {
+        return {
+          status: 400,
+          body: "Please provide both productionDoc and stagingDoc in the request body",
+        };
+      }
+
+      const result = compareDocuments(stagingDoc, productionDoc, ignoredKeys);
+
+      return {
+        status: 200,
+        jsonBody: result,
+      };
+    } catch (error) {
+      return {
+        status: 400,
+        body: "Error processing request: " + error.message,
+      };
+    }
+  }
+});
+
+function compareDocuments(staging, production, ignoredKeys, path = '') {
+    let differences = {};
+    let overallResults = [];
+
+    staging = staging === undefined ? {} : staging;
+    production = production === undefined ? {} : production;
+
+    const keys = new Set([
+        ...Object.keys(staging),
+        ...Object.keys(production)
+    ]);
+
+    for (let key of keys) {
+        const currentPath = path ? `${path}.${key}` : key;
+
+        if (shouldIgnore(currentPath, ignoredKeys)) {
+            continue; // Skip comparison for keys that should be ignored
+        }
+
+        const stagingValue = staging[key];
+        const productionValue = production[key];
+
+        const comparison = compareValues(stagingValue, productionValue, ignoredKeys, currentPath);
+        overallResults.push(comparison.result);
+
+        if (comparison.result !== 'parity' && comparison.differences) {
+            for (let [resultType, diffs] of Object.entries(comparison.differences)) {
+                if (!differences[resultType]) {
+                    differences[resultType] = [];
+                }
+                differences[resultType] = differences[resultType].concat(diffs);
+            }
+        }
+    }
+
+    const overallResult = aggregateOverallResult(overallResults);
+
+    return {
+        overallResult,
+        differences
+    };
+}
+
+function compareValues(val1, val2, ignoredKeys, path) {
```

Comment:
This looks great. 
I would get frustrated having to deal with JS type, null and undefined comparisons 😄